File: leetcode/exercises/2.1.2.medium.two_sum_ii_input_array_is_sorted_1.go
// -----------------------
// Created    : 13/11/2025
// Last Edited: 13/11/2025 
// Topics     : 
// Big O      :
// Problem Id : 167
// -----------------------

package main

import (
    "fmt"
)

func twoSum(numbers []int, target int) []int {
    l := 0
    r := len(numbers) - 1
    sum := 0
    for l < r {
        sum = numbers[l] + numbers[r]
        if sum == target { return []int{l+1, r+1} }
        if sum < target {
            l++
        } else {
            r--
        }
    }
    return []int{0, 0}
}

func main() {
    fmt.Println(twoSum([]int{2,7,11,15},13))
}

File: leetcode/exercises/2.1.2.medium.two_sum_ii_input_array_is_sorted_1.js
// -----------------------
// Created     :3/11/2025
// Last Edited :3/11/2025 
// Topics      :
// Big O       :
// Problem Id  :67
// References  :
// Notes       :
// -----------------------

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    l = 0;
    r = numbers.length - 1;

    while (l < r) {
        let sum = numbers[l] + numbers[r];
        if (sum === target) { return [l+1, r+1]; } 
        if (sum < target) {
            l++;
        } else {
            r--;
        }
    }
    return [0, 0];
};

console.log(twoSum([2,7,11,15],13))
